reference/reference.go

```diff
@@ -236,6 +237,18 @@ func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
 	}, nil
 }
 
+// Match reports whether ref matches the specified pattern.
+func Match(pattern string, ref Reference) bool {
+	matched, err := path.Match(pattern, ref.String())
+	if err != nil {
+		return false
+	}
+	if namedRef, isNamed := ref.(Named); isNamed && !matched {
+		matched, _ = path.Match(pattern, namedRef.Name())
```

Comment:
It just seems confusing to me that a function called `Match` would have this behavior. Could we call it something like `MatchRefOrName`? Feel free to suggest something better.